### PR TITLE
Don't use the configurable e-mail address for Debian SSO

### PR DESCRIPTION
**Pull request by stefanor**
Currently, users can SSO in, which creates an account. Then they change their e-mail address (e.g. from an ugly alioth one, to their canonical address) and the next time they SSO, they get a new account.

Rather, let's keep a wafer account associated with a Debian identity, for life.

Storing this in the Registration group is a bit hacky, but this information is useful to the registration team. It proves account ownership.

### CODE REVIEW

File: wafer/registration/sso.py [hunk 1]
```diff
@@ -129,11 +133,22 @@ def debian_sso(meta):
...
+
+    user = None
+    for kv in KeyValue.objects.filter(
+            group=group, key='debian_sso_email', value=email,
+            userprofile__isnull=False):
+        if kv.userprofile_set.count() > 1:
+            message = 'Multiple accounts have Debian SSOed with address %s'
+            log.warning(message, email)
+            raise SSOError(message % email)
+        user = kv.userprofile_set.first().user
```
Comment (hodgestar) [inline wafer/registration/sso.py]: Will this raise a non-SSO error if there are no matching profiles?

Comment (stefanor) [inline wafer/registration/sso.py]: We've already enforced profiles, in the query (line 141)

Comment (hodgestar) [inline wafer/registration/sso.py]: I'm guessing that the `filter` prevents that from happening.

**Comment by hodgestar** [comment]
:+1:

**Pull request merged**